### PR TITLE
MC-32285: Cannot assign more than 20 sources to product

### PR DESCRIPTION
**Pull request by arnobsh**
### Description (*)

Cannot assign more than 20 sources to product

### Fixed Issues (if relevant)

1. https://jira.corp.magento.com/browse/MC-32285

### STEPS TO REPLICATE:
1) Go to Admin Panel. Catalog > Products.
2) Go to one of those products.
3) Go to the Sources section.
4) Assign more than 20 sources to a product.
5) Save the product.

### EXPECTED RESULTS:
Save product successfully with all assigned sources

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by sidolov** [comment]
@arnobsh please, submit PR to 1.2-develop branch since it's a branch for the next releases. 1.1-develop is for security and critical bugfix only.

File: InventoryLowQuantityNotificationAdminUi/Observer/ProcessSourceItemConfigurationsObserver.php [hunk 1]
```diff
@@ -95,11 +95,49 @@ public function execute(EventObserver $observer)
...
+     * @param array $assignedSources
+     * @param array $stockData
+     * @return array
+     */
+    private function updateAssignedSources(array $assignedSources, array $stockData): array
+    {
+        foreach ($assignedSources as $key => $source) {
+            $isSourceUpdated = false;
+            if (!key_exists('quantity', $source) && isset($source['qty'])) {
+                $source['quantity'] = (int) $source['qty'];
```
Comment (sidolov) [inline InventoryLowQuantityNotificationAdminUi/Observer/ProcessSourceItemConfigurationsObserver.php]: Please, use `$assignedSources[$key]['quantity']` instead and remove `$isSourceUpdated` flag usage in all cases.

**Review by sidolov — CHANGES_REQUESTED** [review]

File: InventoryLowQuantityNotificationAdminUi/Observer/ProcessSourceItemConfigurationsObserver.php (hunk 1 above)
Comment (o-iegorov) [reply InventoryLowQuantityNotificationAdminUi/Observer/ProcessSourceItemConfigurationsObserver.php]: Redundant code refactored

**Review by sidolov — APPROVED** [review]

**Comment by sidolov** [comment]
@magento run all tests

**Pull request merged**